### DISCUSSION TITLE
`path_elements` up to project root bot to fs root

### DISCUSSION THREAD

**Discussion by ghost** [SCons/scons#4628] [Q&A] [answered]
Hello there is `path_elements` property. Is there an easy and sound way to get only path_elements up to project dir? For example I have file `/storage/projects/MyProj/src/file.c` If my main file is placed in `MyProj` I will get list which also includes `/` `/storage/`, `storage/projects`. I can easily filter them out because I already know where is `#` but it is not very good. Is there an easy and clean way to get only `MyProj`, `MyProj/src/`, `MyProj/src/file.c`?

**Comment by mwichmann** [comment]
Could you provide a little more context? What is it you want path_elements for?

**Reply by ghost** [reply]
I have a file `#src/dir/file.c` and I want to get list of parents. I want to get following `['src', 'src/dir']` + optionally in the end file itself `src/dir/file.c` but `path_elemnts` returns all the dirs up to the root of fs and I get `['/', '/projects', '/projects/MyProj', '/projects/MyProj/src', '/projects/MyProj/src/dir']`. I can iterate all over the list and filter paths I don't need because I can compare with "#" path but it is ugly

**Reply by mwichmann** [reply, marked as answer]
`path_elements` (note the more official way is to use the `get_path_elements()` method, though neither are considered "published API") works additively - each node created adds itself to the list it got from its parent directory, which it got after amending its parent's, so on all the way up to the root; there's no filtering capability or argument to return a subset of that.

If you just want the strings, you can use a relative path, make a pathlib path out of it, and use its `parts` attribute - but you'll still have to add the pieces together - that is, 
```
pathlib.PurePath('src/dir/file.c')  --> ('src', 'dir', 'file.c')  # and not ("src", "src/dir", etc.
```
which is probably no simpler than just doing your own filter on the result of `get_path_elements()`.

**Reply by ghost** [reply]
I also though about `parts` and it perfectly worked when I didn't need to concatenate them but yeah in this case it doesn't differ much from filtering... Isn't published? How to determine which is published and what is not?

**Comment by mwichmann** [comment]
"published" means it's in the manpage.  not published doesn't mean you can't use it - it's just a Python program, after all - but it doesn't come with promises of consistency, documentation, etc.
